test/stress/pool-create-hanging.js

```diff
@@ -0,0 +1,193 @@
+/**
+ * Test case for figuring out robust way to recognize if connection is dead
+ * for mysql based drivers.
+ */
+const toxiproxy = require('toxiproxy-node-client')
+const toxicli = new toxiproxy.Toxiproxy('http://localhost:8474')
+const rp = require('request-promise-native')
+
+const mssql = require('../../' + process.env.TEST_DRIVER)
+const debug = require('debug')('mssql-stress')
+
+function delay (ms) {
+  return new Promise((resolve, reject) => {
+    setTimeout(resolve, ms)
+  })
+}
+
+function cloneDeep (val) {
+  return JSON.parse(JSON.stringify(val))
+}
+
+// initial dummy object, which will be overridden by connection process
+let mssqlCon = { connected: false }
+let connectionNumber = 0
+
+async function mssqlQuery (sql) {
+  // recreate connection is not connected / connecting stus
+  if (!mssqlCon.connected && !mssqlCon.connecting) {
+    debug('========== Reconnecting mssql')
+
+    // this test needs toxiproxy to run so docker mssql is also required
+    mssqlCon = new mssql.ConnectionPool({
```

Comment:
does the config belong inside the test itself?

Reply:
Yes, this time configuration should  be hardcoded, because it must match the docker-compose setup which is hardcoded as well (but in yaml language instead of js / json).